app/main.py

```diff
@@ -1,3 +1,61 @@
+from typing import Any
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Any) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        else:
+            return Distance(self.km + other)
```

Comment:
When adding a non-Distance object, ensure that the `other` parameter is a numeric type to prevent runtime errors. Consider adding a type check for `other` to ensure it's an `int` or `float`.